src/devtools/client/debugger/src/components/SourceOutline/SourceOutline.tsx

```diff
@@ -200,12 +177,14 @@ export default function SourceOutlineWrapper() {
   const cursorPosition = useAppSelector(getCursorPosition);
   const selectedSource = useAppSelector(getSelectedSource);
   const replayClient = useContext(ReplayClientContext);
+  const { range: focusRange } = useContext(FocusContext);
 
   let symbols: getSourceOutlineResult | null = null;
   const symbolsCacheValue = useImperativeCacheValue(
-    sourceOutlineCache,
+    outlineHitCountsCache,
     replayClient,
-    selectedSource?.id
+    selectedSource?.id,
+    focusRange ? toPointRange(focusRange) : null
```

Comment:
Because `OutlineHitCountsCache` is a regular cache (not an interval one) I think this is going to cause us to over-fetch in many cases?

e.g.

1. User opens Replay (no focus)
2. We fetch values for the whole recording
3. User sets focus
4. We re-fetch values for the new focus range (even though we've already fetched them)

It's not awful but I wonder if you considered using an interval cache for this? I haven't thought much about it so maybe it wouldn't make sense in this case.

Reply:
Ah, okay that makes sense.